### PR TITLE
Add tests for density-corrected size

### PR DESCRIPTION
**Pull request by noamr**
These test include images that have EXIF values that specify density-corrected size
And they assert that the intrinsic size for those images is corrected by that EXIF.

WhatWG PR: https://github.com/whatwg/html/pull/5574

### CODE REVIEW

**Comment by annevk** [comment]
How does Content-DPR relate to this? I thought this was only about EXIF?

What would be interesting to see:

* Coverage for a wide variety of popular image formats
* Out-of-range values for the EXIF fields.
* Either generating the image EXIF data dynamically or documenting clearly what the EXIF metadata is supposed to be and how it can be inspected.

**Comment by noamr** [comment]
> How does Content-DPR relate to this? I thought this was only about EXIF?`
Oops, probably a copy-paste thing (this is based on old content-dpr tests, as it has a similar purpose)

> 
> What would be interesting to see:
> 
> * Coverage for a wide variety of popular image formats
> * Out-of-range values for the EXIF fields.
Sure, I can add that.

> * Either generating the image EXIF data dynamically or documenting clearly what the EXIF metadata is supposed to be and how it can be inspected.
yea I can try to integrate exiftool into WPT-runner... let's see how that works

**Comment by noamr** [comment]
New patch uses piexif.js to generate density-corrected size images on the fly.
Leaving PNG and other formats for later (after the spec gets some thumbs up) as it's more difficult to implement - some EXIF-related tools only support JPEG.
Also, the spec now does not define ranges, and it is impossible to generate negative numbers in EXIF.

File: density-size-correction/density-corrected-natural-size.html [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    }
+
+    async function test() {
+        await test_valid({width: 10, height: 20, preferredWidth: 20, preferredHeight: 40, resolutionX: 36, resolutionY: 36, resolutionUnit: 2})
+        await test_valid({width: 10, height: 20, preferredWidth: 2, preferredHeight: 4, resolutionX: 360, resolutionY: 360, resolutionUnit: 2})
+        await test_valid({width: 10, height: 20, preferredWidth: 20, preferredHeight: 10, resolutionX: 36, resolutionY: 144, resolutionUnit: 2})
+        await test_valid({width: 10, height: 20, preferredWidth: 10, preferredHeight: 40, resolutionX: 72, resolutionY: 36, resolutionUnit: 2})
+        await test_valid({width: 30, height: 30, preferredWidth: 90, preferredHeight: 30, resolutionX: 24, resolutionY: 72, resolutionUnit: 2})
+
+        await test_valid({width: 10, height: 20, preferredWidth: 20, preferredHeight: 40, resolutionX: 36, resolutionY: 36, resolutionUnit: 2})
```
Comment (yoavweiss) [inline density-size-correction/density-corrected-natural-size.html:27]: What's the difference between those 4 tests?

**Review by yoavweiss — COMMENTED** [review]
Tests look good! I want someone more tied to the project to verify it's OK to land third party code.

**Comment by yoavweiss** [comment]
@foolip - can you take a look at the third-party code here and make sure this is the right way to land it?

File: density-size-correction/density-corrected-natural-size.html (hunk 1 above)
Comment (noamr) [reply density-size-correction/density-corrected-natural-size.html:27]: lores, hires, only one of the dimension, change in aspect ratio.

Comment (noamr) [reply density-size-correction/density-corrected-natural-size.html:27]: oops, forgot to erase some identical tests

**Comment by foolip** [comment]
@yoavweiss how to include third party JS is the topic of https://github.com/web-platform-tests/rfcs/issues/46, which isn't resolved yet, but I'd recommend following the suggestion in https://github.com/web-platform-tests/rfcs/issues/46#issuecomment-605098405. That is, put it in a `third_party` directory, and make sure to include a LICENSE file.

**Comment by noamr** [comment]
Already merged as part of Chromium CL.

**Pull request closed without merging**